packages/paste-core/components/combobox/stories/index.stories.tsx

```diff
@@ -0,0 +1,158 @@
+  .addDecorator(withKnobs)
+  .add('Combobox', () => {
+    const [inputItems, setInputItems] = React.useState(iconItems);
+    return (
+      <>
+        <Combobox
+          autocomplete
+          items={inputItems}
+          labelText="Choose a component:"
+          helpText="This is the help text"
+          optionTemplate={(item: any) => (
+            <MediaObject verticalAlign="center">
+              {item.iconLeft ? (
+                <MediaFigure spacing="space20">
+                  <InformationIcon decorative={false} title="information" />
+                </MediaFigure>
+              ) : null}
+
+              <MediaBody>{item.label}</MediaBody>
+              {item.iconRight ? (
+                <MediaFigure spacing="space20">
+                  <InformationIcon decorative={false} title="information" />
+                </MediaFigure>
+              ) : null}
+            </MediaObject>
+          )}
+          onInputValueChange={({inputValue}) => {
+            if (inputValue !== undefined) {
+              setInputItems(
+                _.filter(iconItems, (item: any) => item.label.toLowerCase().startsWith(inputValue.toLowerCase()))
+              );
+            }
+          }}
+          itemToString={item => (item ? item.label : null)}
+        />
+      </>
+    );
+  })
+  .add('Combobox - Autocomplete', () => {
+    const [inputItems, setInputItems] = React.useState(items);
+    return (
+      <>
+        <Combobox
+          autocomplete
+          items={inputItems}
+          helpText="This is the help text"
+          labelText="Choose a component:"
+          onInputValueChange={({inputValue}) => {
+            if (inputValue !== undefined) {
+              setInputItems(items.filter(item => item.toLowerCase().startsWith(inputValue.toLowerCase())));
+            }
+          }}
+        />
+      </>
+    );
+  })
+  .add('Combobox - Required', () => {
+    return (
+      <>
+        <Combobox items={items} labelText="Choose a component:" helpText="This is the help text" required />
+      </>
+    );
+  })
+  .add('Combobox - Error', () => {
+    return (
+      <>
+        <Combobox items={items} labelText="Choose a component:" helpText="This is the help text" hasError />
+      </>
+    );
+  })
+  .add('Combobox - Disabled', () => {
+    return (
+      <>
+        <Combobox items={items} labelText="Choose a component:" helpText="This is the help text" disabled />
+      </>
+    );
+  })
+  .add('Combobox - insert before and after', () => {
+    return (
+      <>
+        <Combobox
+          items={items}
+          insertBefore={<div>$10.99</div>}
+          insertAfter={
+            <Anchor href="#" display="flex">
+              <InformationIcon decorative={false} title="Get more info" />
+            </Anchor>
+          }
+          labelText="Choose a component:"
+          helpText="This is the help text"
+        />
+      </>
+    );
+  })
+  .add('Combobox - Object', () => {
```

Comment:
These are some really nice examples showing the power and flexibility of combo box

Reply:
I added one more story showing a combobox with a custom(unstyled) input just to make sure the exported components were working.